### PR TITLE
Konvert NetworkRequestHandler

### PR DESCRIPTION
**Pull request by jrodbx**

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso3/NetworkRequestHandler.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          builder.noCache()
+        }
+        if (!shouldWriteToDiskCache(networkPolicy)) {
+          builder.noStore()
+        }
+        builder.build()
+      }
+    }
+
+    val uri = checkNotNull(request.uri, "request.uri == null")
```
Comment (Egorand) [inline picasso/src/main/java/com/squareup/picasso3/NetworkRequestHandler.kt:111]: nit: use Kotlin's native `checkNotNull` that doesn't allocate the error message String unless it's needed?

File: picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      requests.add(request)
+      try {
+        PremadeCall(request, responses.takeFirst())
+      } catch (e: InterruptedException) {
+        throw AssertionError(e)
+      }
+    }
+  }
+
+  @Test @Throws(Exception::class)
```
Comment (Egorand) [inline picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt:70]: nit: are these `@Throws` annotations useful? In Java I believe they were needed to satisfy the compiler, but Kotlin doesn't care. Or are they worth keeping for documentation purposes?

File: picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    hunter.hunt()
+    assertThat(requests.takeFirst().cacheControl.toString())
+      .isEqualTo(CacheControl.FORCE_CACHE.toString())
+  }
+
+  @Test fun shouldRetryTwiceWithAirplaneModeOffAndNoNetworkInfo() {
+    val action = TestUtils.mockAction(URI_KEY_1, URI_1)
+    val cache = PlatformLruCache(0)
+    val hunter = BitmapHunter(picasso, dispatcher, cache, action, networkHandler)
+    assertThat(hunter.shouldRetry(false, null)).isTrue()
```
Comment (Egorand) [inline picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt:106]: nit: if possible, using named args here would help improve readability

File: picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+  @Test fun shouldHandleSchemeInsensitiveCase() {
+    val schemes = arrayOf("http", "https", "HTTP", "HTTPS", "HTtP")
+    for (scheme in schemes) {
+      val uri = URI_1.buildUpon().scheme(scheme).build()
+      assertThat(networkHandler.canHandleRequest(TestUtils.mockRequest(uri))).isTrue()
+    }
+  }
+
+  private companion object {
```
Comment (Egorand) [inline picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt:203]: nit: doesn't really have to be inside a companion

**Review by Egorand — APPROVED** [review]

File: picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt (hunk 2 above)
Comment (jrodbx) [reply picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.kt:70]: good point! i'll clean that up along with @JvmStatic, etc. once everything is converted.

**Pull request merged**